### PR TITLE
Finish LeagueStats and begin TeamStats

### PR DESCRIPTION
**Pull request by hillstew**

### CODE REVIEW

File: lib/team_statistics.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      end
+      season_win_avg[season] = ((home_wins.length.to_f + away_wins.length) / games.length).round(2)
+    end
+    season_win_avg.key(season_win_avg.values.max_by { |value| value})
+  end
+
+  def worst_season(team_id)
+    team_id = team_id.to_i
+    filtered_games = []
+    games_by_team = @games.each do |game_id, game|
```
Comment (lrs8810) [inline lib/team_statistics.rb]: We aren't calling the games_by_team variable anywhere in the worst_season method so we can remove it.

File: lib/team_statistics.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      "team_name" => found_team.team_name,
+      "abbreviation" => found_team.abbreviation,
+      "link" => found_team.link
+    }
+  end
+
+  def best_season(team_id)
+    team_id = team_id.to_i
+    filtered_games = []
+    games_by_team = @games.each do |game_id, game|
```
Comment (lrs8810) [inline lib/team_statistics.rb]: We aren't calling the games_by_team variable anywhere in the best_season method so we can remove it.

**Review by lrs8810 — CHANGES_REQUESTED** [review]
@hillstew Can we remove the games_by_team variable since we aren't calling on it in either method?

File: lib/league_statistics.rb [hunk 3]
```diff
@@ -103,4 +103,144 @@ def worst_defense
       worst_defense == team_id
     end[worst_defense].team_name
   end
+
+  def highest_scoring_visitor
+    games_by_team = @game_teams.group_by do |game_team|
+      game_team.team_id
+    end
```
Comment (grwthomps) [inline lib/league_statistics.rb:110]: Looks like we do this:
```
games_by_team = @game_teams.group_by do |game_team|
   game_team.team_id
end
```
9 different times so we could refactor by moving it out to a helper method.

**Review by grwthomps — CHANGES_REQUESTED** [review]

File: lib/team_statistics.rb (hunk 1 above)
Comment (hillstew) [reply lib/team_statistics.rb]: @lrs8810, yes good point! I will make the change!

File: lib/league_statistics.rb (hunk 3 above)
Comment (hillstew) [reply lib/league_statistics.rb:110]: I agree! I am going to hold off on this change until we are through interation 5 as discussed earlier.

**Review by lrs8810 — APPROVED** [review]
Thank for taking out the games_by_team variables.

**Review by tyladevon — APPROVED** [review]
Reviewed changes to league_statistics, stat_tracker, stat_tracker test, and team_statistics.

**Pull request merged**